rasa/core/actions/action_run_slot_rejections.py

```diff
@@ -65,46 +63,63 @@ async def run(
 
         slot_value = tracker.get_slot(slot_name)
 
-        current_context = dialogue_stack.current_context()
-        current_context[slot_name] = slot_value
-
-        structlogger.debug("run.predicate.context", context=current_context)
-        document = current_context.copy()
-
-        for rejection in top_frame.rejections:
-            condition = rejection.if_
-            utterance = rejection.utter
-
-            try:
-                rendered_template = Template(condition).render(current_context)
-                predicate = Predicate(rendered_template)
-                violation = predicate.evaluate(document)
+        if top_frame.rejections:
```

Comment:
Yes, users can define a rejection for a categorical slot,
Like: 
```- collect: recurrent_payment_type
        rejections:
          - if: not ({"direct debit" "standing order"} contains recurrent_payment_type)
            utter: utter_invalid_recurrent_payment_type```
The rejection mechanism will override the built-in check and I feel it will be the desired behavior.
I suppose a user would define rejections only when they want to define something more than the built-in check has to offer (in the context of categorical slots)